lib/moribus/extensions/has_current_extension.rb

```diff
@@ -6,15 +6,27 @@ module HasCurrentExtension
       # Sets 'is_current' flag of overridden record to false, instead
       # of deleting it or setting foreign key to nil.
       def remove_target!(*)
+        # Use custom update to avoid running ActiveRecord optimistic locking
+        # and to avoid updating lock_version column:
+        klass = target.class
+
         if target.new_record?
           target.is_current = false
+          target.updated_at = Time.zone.now if klass.column_names.include?("updated_at")
         else
-          # Use custom update to avoid running ActiveRecord optimistic locking
-          # and to avoid updating lock_version column:
-          klass = target.class
-
           sql =  "UPDATE #{klass.quoted_table_name} " \
                  "SET \"is_current\" = #{klass.connection.quote(false)} "
+
+          if klass.column_names.include?("updated_at")
+            now = case ActiveRecord::Base.connection.adapter_name
+                  when "SQLite"     then "datetime('now')"
+                  # :nocov:
+                  when "PostgreSQL" then "NOW()"
```

Comment:
Why are you using the database timestamp instead of `Time.zone.now`? Is that not going to potentially wreak confusion with unit tests that use Timecop or similar?

Reply:
no reason, fixed